src/python/doc/rips_complex_tflow_itf_ref.rst

```diff
@@ -0,0 +1,39 @@
+:orphan:
+
+.. To get rid of WARNING: document isn't included in any toctree
+
+TensorFlow layer for Vietoris-Rips persistence
+##############################################
+
+.. include:: differentiation_sum.inc
+
+Example of gradient computed from Vietoris-Rips persistence
+-----------------------------------------------------------
+
+.. testcode::
+
+    from gudhi.tensorflow import RipsLayer
+    import tensorflow as tf
+
+    X = tf.Variable([[1.,1.],[2.,2.]], dtype=tf.float32, trainable=True)
+    rl = RipsLayer(maximum_edge_length=2., dimensions=[0])
+
+    with tf.GradientTape() as tape:
+        dgm = rl.call(X)[0][0]
+        loss = tf.math.reduce_sum(tf.square(.5*(dgm[:,1]-dgm[:,0])))
+
+    grads = tape.gradient(loss, [X])
+    print(grads[0].numpy())
+
+.. testoutput::
+
+    [[-0.5 -0.5]
+     [ 0.5  0.5]]
```

Comment:
Running it on my computer (tensorflow 2.8.0 if it matters), I get a test failure

    Expected:
        [[-0.5 -0.5]
         [ 0.5  0.5]]
    Got:
        [[-0.49999997 -0.49999997]
         [ 0.49999997  0.49999997]]

Reply:
Argh, I updated to TF 2.8.0 but still get
    [[-0.5 -0.5]
     [ 0.5  0.5]]
on my machine. I don't know where this could come from...

Reply:
Maybe we could add, as hidden code so it doesn't show in the generated html, `import numpy` and `numpy.set_printoptions(precision=4)` (or something equivalent).

Reply:
Do you mean something like:

`.. testcode::`
`    :hide:`
`    import numpy`
`    numpy.set_printoptions(precision=4)`

`.. testcode::`

`    from gudhi.tensorflow import RipsLayer`
`    import tensorflow as tf`
`[...]`

?

Reply:
Either that or `testsetup`, yes. But I haven't tried it...

Reply:
OK, just made a push with `testsetup`. I hope this will work.

Reply:
This seems to cause random (they don't happen at every run :confused: ) errors in other test code, for instance
```
File "representations.rst", line 45, in default
Failed example:
    import numpy as np
    from gudhi.representations import Landscape
    # A single diagram with 4 points
    D = np.array([[0.,4.],[1.,2.],[3.,8.],[6.,8.]])
    diags = [D]
    l=Landscape(num_landscapes=2,resolution=10).fit_transform(diags)
    print(l)
Expected:
    [[1.02851895 2.05703791 2.57129739 1.54277843 0.89995409 1.92847304
      2.95699199 3.08555686 2.05703791 1.02851895 0.         0.64282435
      0.         0.         0.51425948 0.         0.         0.
      0.77138922 1.02851895]]
Got:
    [[1.0285 2.057  2.5713 1.5428 0.9    1.9285 2.957  3.0856 2.057  1.0285
      0.     0.6428 0.     0.     0.5143 0.     0.     0.     0.7714 1.0285]]
```
Maybe it also requires a `testcleanup` to reset the precision to its default value (8?) after the tensorflow testcode, so it doesn't interfere with the rest.

Reply:
Like this:
`[...]`
`     print(grads[0].numpy())`
 
`.. testcleanup::`

` .. testoutput::`
 
`     [[-0.5 -0.5]`
`[...]`
?

Reply:
With `numpy.set_printoptions(precision=8)` (or a better way to reset things if there is) in the testcleanup block, yes.
It feels like a hack, but I don't have many other ideas...

Reply:
OK thanks, just made a new push.